### PR TITLE
espressif/esp32/esp32s2/esp32c3: Disable wdt on initialization.

### PR DESCRIPTION
**Pull request by saramonteiro**
## Summary

This MR aims to:
* Fix esp32s2 wdt disabling and wrap it under a function.
* Wrap wdt disabling in esp32.
* Implement wdt disabling in esp32c3.

## Impact

From now on, it's not necessary to disable WDT in bootloader. Start initialization will take care of disabling wdt that was enabled by bootloader.

## Test

I used a bootloader with RWDT enabled for the 3 SoCs and I commented the function that disables it and could see the chips rebooting then I uncommented and the nsh was kept.

### CODE REVIEW

File: arch/risc-v/src/esp32c3/hardware/esp32c3_rtccntl.h [hunk 1]
```diff
@@ -49,13 +49,13 @@
  * write-enable the wdt registers
  */
 
-#define RTC_CNTL_WDT_WKEY_VALUE 0x50D83AA1
+#define RTC_CNTL_WDT_WKEY_VALUE     0x50D83AA1
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/hardware/esp32c3_rtccntl.h:52]:
```suggestion
#define RTC_CNTL_WDT_WKEY_VALUE     0x50d83aa1
```
It is common practice to use lowercase letters for hex literals.

File: arch/risc-v/src/esp32c3/hardware/esp32c3_rtccntl.h [hunk 2]
```diff
@@ -49,13 +49,13 @@
...
 
 /* The value that needs to be written to RTC_CNTL_SWD_WPROTECT_REG
  * to write-enable the wdt registers
  */
 
-#define RTC_CNTL_SWD_WKEY_VALUE 0x8F1D312A
+#define RTC_CNTL_SWD_WKEY_VALUE     0x8F1D312A
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/hardware/esp32c3_rtccntl.h:58]:
```suggestion
#define RTC_CNTL_SWD_WKEY_VALUE     0x8f1d312a
```
It is common practice to use lowercase letters for hex literals.

Comment (saramonteiro) [reply arch/risc-v/src/esp32c3/hardware/esp32c3_rtccntl.h]: Good catch! Solved.

**Review by gustavonihei — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
CI for macOS(sim) fail to download the artifact, since this patch doesn't relate to sim or macOS, let's ignore the CI issue and merge it directly.

**Pull request merged**